Standardlize Tools Code

zhenbinwu opened the pull request:
Split the functions and objects declaration into .cc file from the header file.
Keep object more isolated. In particular, user should create his/her own 
baseline class in their old analysis code.

Commit:
Merge

Commit:
Merge branch 'master' into ZhenbinLocal

Commit:
Update Makefile

1. remove basicCheck, which used global pointer
2. Compile all objects

Commit:
Making baselineDef more objects like

1. split definition and declaration to .h and .cc file
2. Move common input labels as private members
3. Ideally, each analysis jsut need to define their own baseline class
and modify it locally, instead of all messing up this code
4. add nTop objects to the baseline, so that we can get rid of the global
type3ptr, which is dangerous

Commit:
Update customize

Similar idea as baseline changes

Commit:
Silent out the warning of reading filename

Commit:
Update on NtupleReader

pastika commented:
@fslacroix @weihua19900704 Do either of you see any issues interfacing these changes with your code?

@zhenbinwu Hongxuan and I would like you to ensure that the baselineCheck.C file compiles and runs as expected. Then Hongxuan will confirm that these changes do not effect the cut flows.  

My own comments:
- Remove the 2 binaries which were added to the repository
- Remove change to samples.cc

lihux25 commented:
@zhenbinwu would be good if you could have the basicCheck.C modified so that it works in your new modification (if it's done, please confirm this). Then I would like to cross-check the cut flow numbers do not change before and after your modification. Your modification might not change the baseline cuts, but looks to me that there are quite some substantial difference between what you made and previous version. At current stage, it's very important to ensure that physics (cut flow) does not change with the substantial modification. Secondly, ideally the change should involve minimum modification of other users' code...

weihua19900704 commented:
Hi Joe:

For QCD, both running and compiling without problem(get same result as before)
For LL, compiling fine, not running on the full sample, since it takes too long...

Thanks!
Hua
________________________________
From: pastika [notifications@github.com]
Sent: 05 November 2015 00:43
To: susy2015/SusyAnaTools
Cc: Hua Wei
Subject: Re: [SusyAnaTools] Standardlize Tools Code (#49)


@fslacroix<https://github.com/fslacroix> @weihua19900704<https://github.com/weihua19900704> Do either of you see any issues interfacing these changes with your code?

@zhenbinwu<https://github.com/zhenbinwu> Hongxuan and I would like you to ensure that the baselineCheck.C file compiles and runs as expected. Then Hongxuan will confirm that these changes do not effect the cut flows.

My own comments:

  *   Remove the 2 binaries which were added to the repository
  *   Remove change to samples.cc

—
Reply to this email directly or view it on GitHub<https://github.com/susy2015/SusyAnaTools/pull/49#issuecomment-153906203>.

Commit:
Remove accidentally added binary

Commit:
Update for basicCheck

Commit:
Update basicCheck to enable running

zhenbinwu commented:
@pastika @lihux25 
* Done with baselineCheck.C. It only runs as the old code, not responsible for its own features
* 2 binaries are removed
* The warning in samples.cc is meaningless. The idea of the ntuple is able to run independent of CMSSW and LPC. If the filelist is not find, user just got the empty run. 

@lihux25 
If you just make a comparison with master:Tools/baselineDef.h with ZhenbinLocal:Tools/baselineDef.cc, you can see it only reformat the code. No changes are done within the functions.

@weihua19900704 
That is great! It don't change much for user. Only need to link two additional object files.

lihux25 commented:
@zhenbinwu looks like you made change to basicCheck.C. I will take a look at it. Thanks!

zhenbinwu commented:
@lihux25 Necessary changes are made within basicCheck.C for its proper running. But the code is basically the same, still using global variables etc. Please take a look. Thanks

Commit:
Merge branch 'master' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
Fix a bug for lepton counts

Commit:
Merge branch 'master' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
NTupleReader to jump to particular entry for debug

Commit:
Merge branch 'master' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
Merge branch 'master' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
Merge changes in Baseline for Spring15_V3

Commit:
Update samples.py location

Commit:
Remove Zinv in baseline

It conflicted with my code

Commit:
Standardlize searchBins

Commit:
Delete function for Type3TopTagger

Commit:
Merge tag 'Ana_74X_17Nov2015_v3.0' into ZhenbinLocal

Commit:
Merge branch 'master' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
Merge branch 'master' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
Merge branch 'NTupleMakerFix' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
Add destructor for baselineVessel

Commit:
Revert Joe's changes to NTupleReader

It is safer to throw an error than cast to empty

Commit:
Merge branch 'master' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
Move EventFilter out of baseline

EventFilter should be process per event, while baseline can be
overloaded several times. Move it to an independant function

Commit:
Merge branch 'master' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
Ignore ugly basicCheck code..

Commit:
Merge branch 'master' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
Merge with Ntp_74X_04Dec2015_v4.0

Commit:
Update Makefile for LHAPDF on LPC

Commit:
Don't run PDFuncertaity for data

Commit:
Fixing memory leak of PDFUncertainty

Commit:
Add more funs to NTupleReader

Commit:
Merge branch 'master' of github.com:susy2015/SusyAnaTools into ZhenbinLocal

Commit:
Minor fix for bug

Commit:
Fix Makefile for LHAPDF

Commit:
Merge branch 'ZhenbinLocal' of github.com:susy2015/SusyAnaTools into ZhenbinLocal
